Apromore-Custom-Plugins/Process-Simulation-Info-Logic/src/main/java/org/apromore/processsimulation/service/SimulationInfoService.java

```diff
@@ -134,6 +137,18 @@ public ProcessSimulationInfo transformToSimulationInfo(
         return processSimulationInfo;
     }
 
+    private CalendarModel getActiveCalendarModel(SimulationData simulationData) {
+        CalendarModel activeCalendar;
+        if (simulationData.getCalendarModel() == null) {
+            // If no calendar, default to a 24 x 7 calendar
+            activeCalendar = calendarService.getGenericCalendar();
+        } else {
+            activeCalendar = simulationData.getCalendarModel();
+        }
+
+        return activeCalendar;
+    }
+
```

Comment:
It's better to put a constraint NonNull for calendarModel field in SimulationData rather than checking for error here.  It helps detect errors early and fail fast instead of ignoring it. SimulationData represents the contract between simulation data provider (e.g. PDAnalyst) and consumer (e.g. SimulationInfoService). Pls consider adding more constraints such as Non-Null, Non-Empty for other fields if we think it can help enforce this contract.

Reply:
Wouldn't the code be safer this way though, instead of throwing a null pointer exception, that is incurred by the `@NonNull` annotation?

If a SimulationData object is created with a null calendar, we then need to catch the null pointer exception and either
- Show a message to the user or
- Default to using the 24/7 calendar

Either way I see the need for null handling

Reply:
If there's a processing error in PD (simulation data provider) that makes it provide a null calendar (i.e. no calendar) while it does has a business calendar (i.e. 9-5 calendar), then the above error handling can't detect that error because it automatically "fixes" it by using a different calendar. In this contract (SimulationData), the CalendarModel is not optional (i.e. it can't be null) because the SimulationData comes with other time-related data such as average resource time, etc which is calculated based on the simulation data provider's calendar.

In error handling, it is not necessary to write error handling code for all types of errors/exceptions which could occur everywhere, the code will look more and more complicated. NullPointerException is a run-time error meaning something unexpected and wrong happens in the system. It can be handled via user-friendly message such as "Contact your administrator for help", error logs, and then fixed later.